src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -452,5 +454,101 @@ private async Task<List<string>> ExecuteCommandAsync(IExecutionContext context,
+        }
+
+        public async Task RemoveContainerNetworkAsync(IExecutionContext executionContext, object data)
+        {
+            Trace.Entering();
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+            string containerNetwork = data as string;
+            if (string.IsNullOrEmpty(containerNetwork))
+            {
+                containerNetwork = executionContext.Variables.Get(Constants.Variables.Agent.ContainerNetwork);
+            }
+            if (!string.IsNullOrEmpty(containerNetwork))
+            {
+                executionContext.Output($"Remove container network: {containerNetwork}");
+
+                int removeExitCode = await _dockerManger.DockerNetworkRemove(executionContext, containerNetwork);
+                if (removeExitCode != 0)
+                {
+                    executionContext.Error($"Docker network rm failed with exit code {removeExitCode}");
+                }
+                // Remove docker network from env
+                executionContext.Variables.Set(Constants.Variables.Agent.ContainerNetwork, null);
+            }
+        }
+
+        public async Task StartMultipleContainersAsync(IExecutionContext executionContext, object data)
+        {
+            Trace.Entering();
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+
+#if !OS_WINDOWS
+            executionContext.Debug($"Delete stale container networks from previous jobs");
+            int networkPruneExitCode = await _dockerManger.DockerNetworkPrune(executionContext);
+            if (networkPruneExitCode != 0)
+            {
+                executionContext.Warning($"Delete stale container networks failed, docker network prune fail with exit code {networkPruneExitCode}");
+            }
+#endif
+            // Clean up containers left by previous runs
+            executionContext.Debug($"Delete stale containers from previous jobs");
+            var staleContainers = await _dockerManger.DockerPS(executionContext, $"--all --quiet --no-trunc --filter \"label={_dockerManger.DockerInstanceLabel}\"");
+            foreach (var staleContainer in staleContainers)
+            {
+                int containerRemoveExitCode = await _dockerManger.DockerRemove(executionContext, staleContainer);
+                if (containerRemoveExitCode != 0)
+                {
+                    executionContext.Warning($"Delete stale containers failed, docker rm fail with exit code {containerRemoveExitCode} for container {staleContainer}");
+                }
+            }
+
+            List<ContainerInfo> containers = data as List<ContainerInfo>;
+            ArgUtil.NotNull(containers, nameof(containers));
+
+            foreach (var container in containers)
+            {
+                await StartContainerAsync(executionContext, container);
+            }
+        }
+
+        public async Task StopMultipleContainersAsync(IExecutionContext executionContext, object data)
```

Comment:
move public method up and move private method down in the file.

Reply:
StopContainersAsync?

Reply:
Done. I originally wrote Multiple to make it clearly different than the singular one, but since the singular method is now removed from the interface it is not as necessary. I changed the interface to StartContainersAsync and StopContainersAsync